src/metabase/driver/sql/query_processor/boolean_to_comparison.clj

```diff
@@ -0,0 +1,104 @@
+(ns metabase.driver.sql.query-processor.boolean-to-comparison
+  "In Oracle and some other databases, boolean literals cannot appear in the top-level of WHERE clauses or expressions
```

Comment:
Might include the hint about newer Oracle versions supporting booleans here too.